bolt-modules/boltlib/lib/puppet/functions/puppetdb_query.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+require 'bolt/error'
+
+#
+# Makes a query to puppetdb using the bolts puppetdb client.
+#
+Puppet::Functions.create_function(:puppetdb_query) do
+    # This type could be more specific ASTQuery = Array[Variant[String, ASTQuery]]
+  dispatch :make_query do
+    param 'Variant[String, Array[Data]]', :query
+    return_type 'Array[Data]'
+  end
+
+  def make_query(query)
+    puppetdb_client = Puppet.lookup(:bolt_pdb_client) { nil }
+    unless Puppet[:tasks] && puppetdb_client && Puppet.features.bolt?
+      raise Puppet::ParseErrorWithIssue.from_issue_and_stack(
+        Puppet::Pops::Issues::TASK_MISSING_BOLT, action: _('query facts from puppetdb')
+      )
+    end
+
+    puppetdb_client.make_query(query)
```

Comment:
This could have basic specs to make sure it's executable Ruby. I would probably add them to get earlier feedback. But Orchestrator tests will also make sure it's functional.

Reply:
fixed